File: 3. Week 3 - Line Chart/solutions/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.dates as mdates
import os

def visualize():
    data_path = '../../data/trailing_365_distance.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    df = pd.read_csv(data_path)
    df['Date'] = pd.to_datetime(df['Date'])
    
    # 1. Plot
    plt.figure(figsize=(12, 6))
    
    plt.plot(df['Date'], df['TotalDistance'], color='rebeccapurple', linewidth=2, label='Trailing 365d Distance')
    
    # 2. Formatting
    plt.title('Trailing 365-Day Total Distance', fontsize=16)
    plt.xlabel('Date', fontsize=12)
    plt.ylabel('Distance (km)', fontsize=12)
    plt.grid(True, linestyle='--', alpha=0.5)
    
    # 3. Date Axis Formatting
    plt.gca().xaxis.set_major_formatter(mdates.DateFormatter('%Y-%m'))
    plt.gca().xaxis.set_major_locator(mdates.YearLocator())
    plt.xticks(rotation=45)
    
    plt.legend()
    plt.tight_layout()
    
    plt.savefig('trailing_distance_chart.png')
    print("Chart saved to trailing_distance_chart.png")
    plt.show()

if __name__ == "__main__":
    visualize()

File: 3. Week 3 - Line Chart/solutions/r/visualization.R
# Visualize Trailing 365-day Distance using ggplot2

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

data_path <- "../../data/trailing_365_distance.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Ensure Date parsing (readr usually handles YYYY-MM-DD well)
  # But good to be explicit if needed: df$Date <- as.Date(df$Date)
  
  # Plot
  p <- ggplot(df, aes(x = Date, y = TotalDistance)) +
    geom_line(color = "rebeccapurple", linewidth = 1) +
    
    # Scale x axis
    scale_x_date(date_labels = "%Y-%m", date_breaks = "1 year") +
    
    # Labels
    labs(
      title = "Trailing 365-Day Total Distance",
      x = "Date",
      y = "Distance (km)"
    ) +
    
    # Theme
    theme_minimal() +
    theme(
      axis.text.x = element_text(angle = 45, hjust = 1),
      panel.grid.major = element_line(linetype = "dashed"),
      panel.grid.minor = element_blank()
    )
  
  ggsave("trailing_distance_chart.png", plot = p, width = 12, height = 6)
  print("Chart saved to trailing_distance_chart.png")

} else {
  print(paste("Error: File not found at", data_path))
}
